src/control/controllers/PidController.cpp

```diff
@@ -0,0 +1,225 @@
+//
+// Created by mrlukasbos on 29-3-19.
+//
+
+#include <control/controllers/PidController.h>
+#include "utilities/Constants.h"
+
+//**********************************
+//Constructor functions
+//**********************************
+PidController::PidController(double p, double i, double d) : PidController(p,i,d,0){}
+
+PidController::PidController(double p, double i, double d, double f){
+    maxIOutput=0;
+    maxError=0;
+    errorSum=0;
+    maxOutput=0;
+    minOutput=0;
+    setpoint=0;
+    lastActual=0;
+    firstRun=true;
+    reversed=false;
+    outputRampRate=0;
+    lastOutput=0;
+    outputFilter=0;
+    setpointRange=0;
+
+    P=p; I=i; D=d; F=f;
+}
+
+PidController::PidController(std::tuple<double, double, double> pid)
+        : PidController(std::get<0>(pid), std::get<1>(pid), std::get<2>(pid)) {}
+
+PidController::PidController(std::tuple<double, double, double, double> pidf)
+        : PidController(std::get<0>(pidf), std::get<1>(pidf), std::get<2>(pidf), std::get<3>(pidf)) {}
+
+
+//**********************************
+//Configuration functions
+//**********************************
```

Comment:
In the original pid.cpp file we had more documentation here, why did you remove it?